src/ray/common/client_connection.cc

```diff
@@ -201,6 +201,8 @@ Status ServerConnection::ReadBuffer(
       bytes_remaining -= bytes_read;
       if (error.value() == EINTR) {
         continue;
+      } else if (error.value() == ENOENT) {
```

Comment:
Two questions:
- `ENOENT` indicates no such entry, or not found, is `NotFound` a better interpretation? https://github.com/ray-project/ray/blob/7b476ea4fd3f7addc4b87099f94d2d46a379b427/src/ray/common/status.h#L198-L200
- Suggest to apply early return, which is suggested by clang-tidy: https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html

```
if (...) {
  continue;
}
if (...) {
  return ...;
}
// do something
```